rmgpy/rmg/model.py

```diff
@@ -795,7 +798,35 @@ def processNewReactions(self, newReactions, newSpecies, pdepNetwork=None):
                         self.core.reactions.remove(rxn)
                     if rxn in self.edge.reactions:
                         self.edge.reactions.remove(rxn)
-
+            
+            if not numpy.isinf(self.toleranceThermoKeepSpeciesInEdge) and spcs != []: #do thermodynamic filtering
+            
```

Comment:
I agree, I like the idea of separating it as it's own function: thermoFilterSpecies.  I'm a bit torn on the public vs private, but I think it would come out clean enough that it could be called separately in a different context so I would be inclined to make it public.  However, I'm not really sure what other context that would be.